{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: sqlalchemy in c:\\users\\hp support\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (1.4.42)\n",
      "Requirement already satisfied: greenlet!=0.4.17 in c:\\users\\hp support\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from sqlalchemy) (1.1.3.post0)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "[notice] A new release of pip available: 22.2.2 -> 22.3\n",
      "[notice] To update, run: python.exe -m pip install --upgrade pip\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: mysqlclient in c:\\users\\hp support\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (2.1.1)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "[notice] A new release of pip available: 22.2.2 -> 22.3\n",
      "[notice] To update, run: python.exe -m pip install --upgrade pip\n",
      "UsageError: Line magic function `%pip3` not found.\n"
     ]
    }
   ],
   "source": [
    "%pip install -U sqlalchemy\n",
    "from sqlalchemy import create_engine as ce\n",
    "import pandas as pd\n",
    "#para agregar el conector python mysql a su entorno python.\n",
    "%pip install mysqlclient\n",
    "%pip3 install PyMySQL\n",
    "import  pymysql"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: cryptography in c:\\users\\hp support\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (38.0.1)\n",
      "Requirement already satisfied: cffi>=1.12 in c:\\users\\hp support\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from cryptography) (1.15.1)\n",
      "Requirement already satisfied: pycparser in c:\\users\\hp support\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from cffi>=1.12->cryptography) (2.21)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "[notice] A new release of pip available: 22.2.2 -> 22.3\n",
      "[notice] To update, run: python.exe -m pip install --upgrade pip\n"
     ]
    }
   ],
   "source": [
    "!pip3 install cryptography"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: PyMySQL[rsa] in c:\\users\\hp support\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (1.0.2)\n",
      "Requirement already satisfied: cryptography in c:\\users\\hp support\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from PyMySQL[rsa]) (38.0.1)\n",
      "Requirement already satisfied: cffi>=1.12 in c:\\users\\hp support\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from cryptography->PyMySQL[rsa]) (1.15.1)\n",
      "Requirement already satisfied: pycparser in c:\\users\\hp support\\appdata\\local\\programs\\python\\python310\\lib\\site-packages (from cffi>=1.12->cryptography->PyMySQL[rsa]) (2.21)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "[notice] A new release of pip available: 22.2.2 -> 22.3\n",
      "[notice] To update, run: python.exe -m pip install --upgrade pip\n"
     ]
    }
   ],
   "source": [
    "!pip3 install PyMySQL[rsa]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\HP SUPPORT\\AppData\\Local\\Temp\\ipykernel_17412\\2390203845.py:5: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  data = pd.read_csv(r\"tabla_general_ps.csv\", sep=\",\", encoding=\"utf-8\")\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "2222645"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Connect to MySQL\n",
    "mysql_engine = ce(\"mysql+pymysql://eliza:snow14@localhost/preciosdb\")\n",
    "\n",
    "#Import some data from a CSV file\n",
    "data = pd.read_csv(r\"tabla_general_ps.csv\", sep=\",\", encoding=\"utf-8\")\n",
    "\n",
    "#Load the imported CSV into yur database\n",
    "data.to_sql('tabla_general_ps', mysql_engine)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.4 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "c01cf4a2356650b355da24661dd0626acaa1665bc8e66dede1d15fa072b6b8b3"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
